### PR TITLE
Add support for headless Hassbian

### LINKED ISSUE #23: No voice-recognizer.pid file when running main.py
Following the HACKING.md file up to these steps:
``` shell
cd ~/voice-recognizer-raspi
source env/bin/activate
python3 src/main.py
```
This is the traceback I get.
``` shell
Traceback (most recent call last):
  File "src/main.py", line 302, in <module>
    main()
  File "src/main.py", line 139, in main
    create_pid_file(args.pid_file)
  File "src/main.py", line 105, in create_pid_file
    with open(file_name, 'w') as pid_file:
FileNotFoundError: [Errno 2] No such file or directory: '/run/user/1000/voice-recognizer.pid'
```

### PR DESCRIPTION
**Pull request by jordandrako**
These are the changes I had to make in order for this to run on my hassbian install. 

Sorry for making a new pull request, my commits got messy so I nuked and restarted.
Should fix #23

### CODE REVIEW

File: scripts/install-deps.sh [hunk 1]
```diff
@@ -27,7 +27,7 @@ scripts_dir="$(dirname "${BASH_SOURCE[0]}")"
 
 sudo apt-get -y install alsa-utils python3-all-dev python3-pip python3-numpy \
   python3-scipy python3-virtualenv rsync sox libttspico-utils ntpdate
-sudo pip3 install --upgrade pip virtualenv
+sudo pip3 install --upgrade pip virtualenv Rpi.GPIO
```
Comment (axilleas) [inline scripts/install-deps.sh]: @jordandrako I tackled the deps in another PR https://github.com/google/aiyprojects-raspbian/pull/35/files, you can remove the changes from this file :)

File: src/main.py [hunk 2]
```diff
@@ -63,8 +63,16 @@
...
+# Grab pid file info
+USER_ID = '%d' % os.getuid()
+GROUP_ID = '%d' % os.getgid()
+PID_PATH = '/run/user/' + USER_ID
+PID_FILE = PID_PATH + '/voice-recognizer.pid'
+PID_COMMAND = 'sudo mkdir ' + PID_PATH + ' && sudo chown -R ' + USER_ID + ':' + GROUP_ID + ' ' + PID_PATH
+
+# Create and own pid directory if it doesn't exist
+if not os.path.exists(PID_PATH):
+    os.system(PID_COMMAND)
```
Comment (ensonic) [inline src/main.py]: Also this needs two blank lines before the start of the function.

**Comment by axilleas** [comment]
I'm wondering how this will work with sudo since `main.py` is run with the regular user 🤔 

I was thinking maybe we can use [`tmpfiles.d`](https://wiki.archlinux.org/index.php/Systemd#Temporary_files) and create the subdirectory when `install-services.sh` is run. I opened https://github.com/google/aiyprojects-raspbian/pull/46 as a test.

**Comment by drigz** [comment]
Thanks for investigating the issue, @jordandrako and @axilleas! We've discussed this, and we'd prefer to fall back to `/tmp` when `/run/user` doesn't exist, so that it's also portable to distros that don't use systemd. We'll look at making this change this afternoon.

**Pull request closed without merging**